clients/cobol-lsp-vscode-extension/src/services/util/ProfileUtils.ts

```diff
@@ -0,0 +1,54 @@
+import { ZoweVsCodeExtension } from "@zowe/zowe-explorer-api/lib/vscode";
+import * as path from "path";
+import * as vscode from "vscode";
+import { COBOL_CBL_EXT, COBOL_COB_EXT, COBOL_COBOL_EXT, SETTINGS_CPY_SECTION } from "../../constants";
+
+export class ProfileUtils {
+    public static getProfileNameForCopybook(cobolFileName: string): (string | undefined) {
+        const zoweExplorerApi = ZoweVsCodeExtension.getZoweExplorerApi();
+        const availableProfiles = zoweExplorerApi.getExplorerExtenderApi().getProfilesCache().getProfiles();
+        return ProfileUtils.getValidProfileForCopybookDownload(cobolFileName, availableProfiles.map(ele => ele.name));
+    }
+
+    private static getValidProfileForCopybookDownload(programName: string, availableProfiles: string[]): string {
+        const profileFromDoc = ProfileUtils.getProfileFromDocument(programName, availableProfiles);
+        if (!profileFromDoc) {
+            return vscode.workspace.getConfiguration(SETTINGS_CPY_SECTION).get("profiles");
+        }
+        return profileFromDoc;
+    }
+
+    private static getProfileFromDocument(programName: string, availableProfiles: string[]): (string | undefined) {
+        for (const doc of vscode.workspace.textDocuments) {
+            const docPath = doc.fileName;
+            if (!ProfileUtils.isCobolProgram(docPath)) {
```

Comment:
What will happen if the file has the language id "COBOL" and doesn't have the extension?

Reply:
It won't be able to resolve. But as mentioned by @grianbrcom - "we have an agreement that COBOL source must be a member of COBOL data set. So, if you create a dataset like AG891842.COBOL with a member CODE. Zowe explorer downloads member AG891842.COBOL(CODE) as CODE.CBL file"

That's why I kept this check, so as to avoid looking for profiles from unwanted files in the workspace.

Reply:
We have to support loading copybooks also for the files that were added not from Zowe. And they may have no extension at all.

Reply:
Agree. Ok, so it makes sense to remove this check of extension. I will test it once and commit. Thanks :)

Reply:
updated. Works without extension as well